code/modules/mob/dead/observer/orbit.dm

```diff
@@ -17,7 +17,7 @@
 
 	if (action == "orbit")
 		var/ref = params["ref"]
-		var/atom/movable/poi = locate(ref)
+		var/atom/movable/poi = (locate(ref) in GLOB.mob_list) || (locate(ref) in GLOB.poi_list)
 		if (poi != null)

```

Comment:
Calling `locate` two times - not optimal.
```suggestion
		var/atom/movable/poi = locate(ref)
		if (poi != null && (poi in GLOB.mob_list || poi in GLOB.poi_list))

```

Reply:
This would revert back to the behaviour AnturK wanted changed (not limiting the scope of the locate(ref))

The locate(ref) without any scope (ie, without locate(ref) in list) searches globally and returns the atom/movable/ (or null if the ref no longer exists)

I figured that it would be more performant to locate() two smaller lists sequentially as opposed to concatting two list and doing locate() once.

Do you have any further comments based on the above? I tried asking in the Discord but #coding-general was too busy bitching about Families for anyone to respond.

Reply:
Ok, I was not aware of this syntactic feature of locate.